tests/e2e/vsan_max_tkg_wldi.go

```diff
@@ -0,0 +1,315 @@
+		// Get VM name of the host from zone2
+		hostlist := getVmNameFromHostIp([]string{hostIp})
+		gomega.Expect(len(hostlist) > 0).To(gomega.BeTrue())
+		// Power off the ESX host
+		err = vMPowerMgmt(tbinfo.user, tbinfo.location, tbinfo.podname, hostlist, false)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		isHostDown = true
+		defer func() {
+			if isHostDown {
+				err = vMPowerMgmt(tbinfo.user, tbinfo.location, tbinfo.podname, hostlist, true)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				err = waitForHostToBeUp(hostIp, time.Minute*40)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}()
+		err = waitForHostToBeDown(ctx, hostIp)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		// Bring down one of the host from vSAN max fault domain-1
+		fdHostVmName := ""
+		fdHostIp := ""
+		for hostIp, site := range fdMap {
+			if strings.Contains(site, vsanMaxFdName) {
+				fdHostIp = hostIp
+				fdHostVmName = getVmNameFromHostIp([]string{hostIp})
+				err = vMPowerMgmt(tbinfo.user, tbinfo.location, tbinfo.podname, fdHostVmName, false)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				isFdHostDown = true
+				break
+			}
+		}
+		defer func() {
+			if isFdHostDown {
+				err = vMPowerMgmt(tbinfo.user, tbinfo.location, tbinfo.podname, fdHostVmName, true)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				err = waitForHostToBeUp(fdHostIp, time.Minute*40)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}()
+		err = waitForHostToBeDown(ctx, fdHostIp)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		// Check for TKG VM and STS pod status
+		nodeList, err := fnodes.GetReadySchedulableNodes(ctx, f.ClientSet)
+		framework.ExpectNoError(err, "Unable to find ready and schedulable Node")
+		gomega.Expect(len(nodeList.Items) == 6).To(gomega.BeTrue())
+		fss.WaitForStatusReadyReplicas(ctx, client, statefulsetRwo, replicas)
+		fss.WaitForStatusReadyReplicas(ctx, client, statefulsetRwm, replicas)
+
+		// Bring up host from zone-2 and host from fault domain-1
+		err = vMPowerMgmt(tbinfo.user, tbinfo.location, tbinfo.podname, hostlist, true)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		isHostDown = false
+		err = waitForHostToBeUp(hostIp, time.Minute*40)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		err = vMPowerMgmt(tbinfo.user, tbinfo.location, tbinfo.podname, fdHostVmName, true)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		isFdHostDown = false
+		err = waitForHostToBeUp(fdHostIp, time.Minute*40)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		//Bring down all hosts from vsphere cluster in zone-3
+		zone3 := topologyAffinityDetails[topologyCategories[0]][2]
+		nodes = getNodesFromZone(ctx, client, zone3)
+		gomega.Expect(len(nodes) > 0).To(gomega.BeTrue())
+		var nodeIps []string
+		for _, node := range nodes {
+			nodeIps = append(nodeIps, node.Status.Addresses[1].Address)
+		}
+		hostlist = getVmNameFromHostIp(nodeIps)
```

Comment:
I would suggest move this entire code to some util so that others can also use this code logic for future requirement.. even i need to write this logic for vsan max